packages/apps/reputation-oracle/server/src/modules/user/user.controller.ts

```diff
@@ -170,7 +178,16 @@ export class UserController {
   public async prepareSignature(
     @Body() data: PrepareSignatureDto,
   ): Promise<SignatureBodyDto> {
-    return await this.userService.prepareSignatureBody(data.type, data.address);
+    let nonce;
+    if (data.type === SignatureType.SIGNIN) {
+      nonce = (await this.userRepository.findOneByAddress(data.address))?.nonce;
```

Comment:
@portuu3 do you see any reason why we store this in DB and not just generate on UI?

IMU we can get rid of this `prepare-signature` completely on RepO, UI/user needs to generate some "nonce", sign it and send together with signature, then RepO is able to verify it